src/asm/lexer.c

```diff
@@ -767,16 +769,23 @@ static int peekInternal(uint8_t distance)
 	return (unsigned char)lexerState->buf[(lexerState->index + distance) % LEXER_BUF_SIZE];
 }
 
+/* forward declarations for peek */
+static void shiftChars(uint8_t distance);
+static char const *readInterpolation(void);
+
 static int peek(uint8_t distance)
 {
-	int c = peekInternal(distance);
+	int c;
+
+restart:
```

Comment:
I would use recursion or a loop for this instead.

Reply:
@ISSOtm said he preferred `goto restart` to `continue` in a `for (;;)` loop. If tail call optimization can be relied upon, recursive `return peek(distance)` may be neater.

Reply:
I would prefer to avoid relying on tail calling for this.

Reply:
It's reasonable to expect a compiler to perform tail-call elimination on something this simple.

Reply:
We may see to that when we come to a round of performance tuning.